pkg/operator/staticpod/startupmonitor/fallback.go

```diff
@@ -128,7 +136,46 @@ func (f *staticPodFallback) fallbackToPreviousRevision(reason, message string) e
 }
 
 func (f *staticPodFallback) markRevisionGood() error {
-	return f.createLastKnowGoodRevisionFor(f.revision, true)
+	if err := f.createLastKnowGoodRevisionFor(f.revision, true); err != nil {
+		return err
+	}
+
+	if f.operatorClient == nil || len(f.nodeName) > 0 {
+		return nil
+	}
+
+	// the startup-monitor is authorative to signal readiness of a revision. The operator
+	// is waiting for this to happen. Otherwise, the operator could assume readiness while
+	// the startup-monitor falls back, leading to an awkward situation.
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		_, status, rv, err := f.operatorClient.GetStaticPodOperatorState()
+		if err != nil {
+			return err
+		}
+
+		for i := range status.NodeStatuses {
+			s := &status.NodeStatuses[i]
+			if s.NodeName != f.nodeName {
+				continue
+			}
+
+			if int(s.TargetRevision) != f.revision {
+				// target revision mismatch
+				return nil
+			}
+
+			s.TargetRevision = 0
```

Comment:
shouldn't `s.TargetRevision` point to `f.revision` ?

Reply:
Do we have to switch `s.CurrentRevision` to `f.revision` ?

Reply:
Target revision is reset on readiness, i.e. zero.

Reply:
fixed currentRevision.